ZenPacks/zenoss/LinuxMonitor/HardDisk.py

```diff
@@ -80,15 +85,40 @@ def impacted_object(self):
             return lv
 
     def storage_disk_lun(self):
-        # return the UCS storage disk/virtual drive
-        # if disk_ids contains the IDs of disk/virtual drive
+        """Return the storage disk/virtual drive based on disk_ids."""
+        keywords = set()
         if self.disk_ids:
-            results = ICatalogTool(self.getDmdRoot('Devices')).search(
-                query=Or(*[Eq('searchKeywords', id)
-                           for id in self.disk_ids]))
-
-            for brain in results:
-                try:
-                    yield brain.getObject()
-                except Exception:
-                    continue
+            for disk_id in self.disk_ids:
+                # Add disk_id to keywords to save compatibility with UCS storage
+                keywords.add(disk_id)
+
+                ldisk_id = disk_id.lower()
+                if ldisk_id.startswith('wwn-0x'):
+                    keywords.add(
+                        'has-target-wwn:{}'.format(ldisk_id.strip('wwn-0x'))
+                    )
+        return keyword_search(self.getDmdRoot('Devices'), keywords)
```

Comment:
`keyword_search()` returns a generator whereas this `storage_disk_lun()` method used to be a generator. There's a functional difference between the two. I think you need to do the following instead of just returning keyword_search's return value.

```python
for obj in keyword_search(self.getDmdRoot('Devices'), keywords):
    yield obj
```

A of Python 3.3 this would be more simply `yield from keyword_search(self.getDmdRoot('Devices'), keywords)`, but for now we have to explicitly expand the generator.